lib/queuePopulator/LogReader.js

```diff
@@ -322,6 +307,9 @@ class LogReader {
 
     _processLogEntry(batchState, record, entry) {
         function _transformKey(key) {
+            if (!key) {
+                return key;
```

Comment:
I think it's more straightforward to `return null` or `return ''` depending on the convention used